UPGRADE.rst

```diff
@@ -94,11 +94,13 @@ Requirement for X-Forwarded-Proto header
 When using Synapse with a reverse proxy (in particular, when using the
 `x_forwarded` option on an HTTP listener), Synapse now expects to receive an
 `X-Forwarded-Proto` header on incoming HTTP requests. If it is not set, Synapse
-will log a warning on each received request.
+will log a warning on each received request. Synapse also requires the `Host`
+header to be preserved.
```

Comment:
I worry that it will be easy to miss this one sentence. It probably merits its own paragraph (maybe before "See the reverse proxy documentation...").